### ISSUE TITLE
Uncaught TypeError: Arguments to path.join must be strings

### ISSUE THREAD

**Issue by stramel** [AtomLinter/Linter#329]
### Steps to Reproduce

1. Refreshed Atom (ctrl+alt+r)

**Atom Version**: 0.170.0-f151697
**System**: linux 3.13.0-43-generic
**Thrown From**: [linter](https://github.com/AtomLinter/Linter) package, v0.9.1

### Stack Trace

Uncaught TypeError: Arguments to path.join must be strings

```
At path.js:390

TypeError: Arguments to path.join must be strings
  at Object.exports.join (path.js:390:15)
  at LinterJshint.Linter.getNodeExecutablePath (/home/stramel/.atom/packages/linter/lib/linter.coffee:114:10)
  at LinterJshint.Linter.getCmdAndArgs (/home/stramel/.atom/packages/linter/lib/linter.coffee:90:25)
  at LinterJshint.Linter.lintFile (/home/stramel/.atom/packages/linter/lib/linter.coffee:122:24)
  at /home/stramel/.atom/packages/linter/lib/linter-view.coffee:138:18
  at Array.forEach (native)
  at /home/stramel/.atom/packages/linter/lib/linter-view.coffee:137:18
  at Object.oncomplete (fs.js:93:15)

```

### Commands

```
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "welcome",
      "open-on-github",
      "background-tips",
      "language-xml",
      "package-generator",
      "release-notes",
      "selection-count",
      "styleguide",
      "linter-jscs",
      "atom-spotify",
      "autocomplete-paths",
      "atom-color-highlight"
    ],
    "themes": [
      "atom-dark-ui",
      "monokai"
    ],
    "projectHome": "~/Projects/foundationui"
  },
  "linter": {
    "lintOnEditorFocus": false
  }
}
```

### Installed Packages

```coffee
# User
atom-jasmine, v0.4.0
autoclose-html, v0.13.0
autocomplete-plus, v1.1.0
autocomplete-snippets, v0.3.2
file-icons, v1.4.4
highlight-line, v0.9.2
highlight-selected, v0.7.0
linter, v0.9.1
linter-bootlint, v0.0.3
linter-jshint, v0.1.0
linter-less, v0.3.1
monokai, v0.9.0
pain-split, v1.1.2
project-manager, v1.14.1
project-palette-finder, v2.4.7

# Dev
No dev packages
```

/cc @atom/core

**Comment by AgentME** [comment]
This pops up on every startup for me.

**Comment by uncedric** [comment]
+1

**Comment by noseglid** [comment]
+1

**Comment by dmnd** [comment]
Thanks for reporting all, it's fixed in v0.10.0

**Comment by yellottyellott** [comment]
XD
